xbmc/video/windows/GUIWindowVideoBase.cpp

```diff
@@ -915,10 +921,28 @@ void CGUIWindowVideoBase::GetContextButtons(int itemNumber, CContextButtons &but
           (!item->HasProperty("IsPlayable") || item->GetProperty("IsPlayable").asBoolean()) &&
           m_vecItems->Size() > 1 && itemNumber < m_vecItems->Size() - 1)
       {
-        if (!CServiceBroker::GetSettingsComponent()->GetSettings()->GetBool(CSettings::SETTING_VIDEOPLAYER_AUTOPLAYNEXTITEM))
-          buttons.Add(CONTEXT_BUTTON_PLAY_AND_QUEUE, 13412);
-        else
+        int settingValue = SETTING_AUTOPLAYNEXT_UNCATEGORIZED;
+
+        if (item->IsVideoDb() && item->HasVideoInfoTag())
+        {
+          std::string mediaType = item->GetVideoInfoTag()->m_type;
+
+          if (mediaType == MediaTypeMusicVideo)
+            settingValue = SETTING_AUTOPLAYNEXT_MUSICVIDEOS;
+          else if (mediaType == MediaTypeEpisode)
+            settingValue = SETTING_AUTOPLAYNEXT_EPISODES;
+          else if (mediaType == MediaTypeMovie)
+            settingValue = SETTING_AUTOPLAYNEXT_MOVIES;
+        }
+
+        const auto setting = std::dynamic_pointer_cast<CSettingList>(
```

Comment:
I would not use auto here as type CSettingsList* is not really complex. IMO, explicitly naming the type improves readability of the code.

Reply:
@ksooo thanx for the review :-)
fixed the three issues above.

this one was a c&p from Montellese's code, assuming he knows best when it comes to settings stuff:
https://github.com/xbmc/xbmc/blob/4c9f798b44c87b8197c3632f758963728cabb367/xbmc/view/GUIViewState.cpp#L514-L516